packages/jetpack-connection/lib/Manager.php

```diff
@@ -0,0 +1,241 @@
+	 * Should be changed to be protected.
+	 */
+	public function initialize_server() {
+
+	}
+
+	/**
+	 * Checks if the current request is properly authenticated, bails if not.
+	 * Should be changed to be protected.
+	 */
+	public function require_authentication() {
+
+	}
+
+	/**
+	 * Verifies the correctness of the request signature.
+	 * Should be changed to be protected.
+	 */
+	public function verify_signature() {
+
+	}
+
+	/**
+	 * Attempts Jetpack registration which sets up the site for connection. Should
+	 * remain public because the call to action comes from the current site, not from
+	 * WordPress.com.
+	 *
+	 * @return Integer zero on success, or a bitmask on failure.
+	 */
+	public function register() {
+
+	}
+
+	/**
+	 * Set the callable that would be used to generate secrets.
+	 *
+	 * @param Callable a function that returns a secure string to be used as a secret.
+	 */
+	public function set_secret_callable( $callable ) {
+		$this->secret_callable = $callable;
+	}
+
+	/**
+	 * Generates two secret tokens and the end of life timestamp for them.
+	 *
+	 * @param String  $action  The action name.
+	 * @param Integer $user_id The user identifier.
+	 * @param Integer $exp     Expiration time in seconds.
+	 */
+	public function generate_secrets( $action, $user_id, $exp ) {
+		$secret_name = 'jetpack_' . $action . '_' . $user_id;
+		$secret_value = array(
+			'secret_1'  => call_user_func( $this->secret_callable ),
+			'secret_2'  => call_user_func( $this->secret_callable ),
+			'exp'       => time() + $exp,
+		);
+
+		$secrets[ $secret_name ] = $secret_value;
+
+		$this->option_backend->update_option( 'jetpack_secrets', $secrets );
+		return $secrets[ $secret_name ];
+	}
+
+	/**
+	 * Returns two secret tokens and the end of life timestamp for them.
+	 *
+	 * @param String  $action  The action name.
+	 * @param Integer $user_id The user identifier.
+	 * @param Integer $exp     Expiration time in seconds.
+	 * @return string|array an array of secrets or an error string.
+	 */
+	public function get_secrets( $action, $user_id, $exp ) {
+		$secret_name = 'jetpack_' . $action . '_' . $user_id;
+		$secrets = $this->option_backend->get_option( 'jetpack_secrets', array() );
+
+		if ( ! isset( $secrets[ $secret_name ] ) ) {
+			return self::SECRETS_MISSING;
+		}
+
+		if ( $secrets[ $secret_name ]['exp'] < time() ) {
+			$this->delete_secrets( $action, $user_id );
+			return self::SECRETS_EXPIRED;
+		}
+
+		return $secrets[ $secret_name ];
+	}
+
+	/**
+	 * Deletes secret tokens in case they, for example, have expired.
+	 *
+	 * @param String  $action  The action name.
+	 * @param Integer $user_id The user identifier.
+	 * @return string|array an array of secrets or an error string.
+	 */
+	protected function delete_secrets( $action, $user_id ) {
+		$secret_name = 'jetpack_' . $action . '_' . $user_id;
+		$secrets = $this->option_backend->get_option( 'jetpack_secrets', array() );
```

Comment:
Should something like `jetpack_secrets` be a class constant?

Reply:
Thank you, I have implemented that in one of the latest commits.